### PR TITLE
types: add the nvme Cancel command opcode

### PR DESCRIPTION
**Pull request by maurizio-lombardi**
Defined by TP4097

### CODE REVIEW

**Comment by maurizio-lombardi** [comment]
It fails the build here on github but I've just added an enum... must be a problem in the tests.

**Comment by martin-belanger** [comment]
> It fails the build here on github but I've just added an enum... must be a problem in the tests.

Could it be that the new enum is missing documentation?

**Comment by maurizio-lombardi** [comment]
> > It fails the build here on github but I've just added an enum... must be a problem in the tests.
> 
> Could it be that the new enum is missing documentation?

You're probably right, the errors are a bit cryptic but in any case I have to add the new opcode to the docs

**Comment by maurizio-lombardi** [comment]
much better

File: doc/man/nvme_io_opcode.2 [hunk 1]
```diff
@@ -56,6 +56,10 @@ enum nvme_io_opcode {
 , 
 .br
 .br
+.BI "    nvme_cmd_cancel"
+,
+.br
+.br
```
Comment (igaw) [inline doc/man/nvme_io_opcode.2:62]: Please drop the generated doc changes. The docs get generated on release time anyway. Thanks

Comment (maurizio-lombardi) [reply doc/man/nvme_io_opcode.2]: Done, is it ok now?

**Comment by igaw** [comment]
Thanks! 

Should we also add the rest of the TP? Not sure if it would be useful from userspace.

**Comment by maurizio-lombardi** [comment]
> Thanks!
> 
> Should we also add the rest of the TP? Not sure if it would be useful from userspace.

I am going to add some parts to nvme-cli because I have to fix things like this:

`
# nvme effects-log /dev/nvme0n1 -H
[...]
I/O Commands
IOCS9    [Dataset Management              ] 00000001  CSUPP+  LBCC-  NCC-  NIC-  CCC-  USS-  No command restriction
IOCS24   [Unknown                         ] 00000001  CSUPP+  LBCC-  NCC-  NIC-  CCC-  USS-  No command restriction
IOCS121  [Zone Management Send            ] 00000003  CSUPP+  LBCC+  NCC-  NIC-  CCC-  USS-  No command restriction

`

See this "Unknown" thing? It's the cancel command

**Pull request merged**